src/main/java/com/autotune/experimentManager/core/ExperimentTrialHandler.java

```diff
@@ -153,27 +157,69 @@ public void startExperimentTrials() {
                         kubernetesServices
                 );
                 IntStream.rangeClosed(1, numberOFIterations).forEach(
-                        i -> {
-                            deploymentHandler.initiateDeploy();
-                            //Check if deployment is ready
-                            int deploymentIsReadyWithinMinute = EMConstants.TimeConv.DEPLOYMENT_IS_READY_WITHIN_MINUTE;
-                            long millisMinutes = TimeUnit.MINUTES.toMillis(deploymentIsReadyWithinMinute);
-                            int count = (int)millisMinutes/EMConstants.TimeConv.DEPLOYMENT_CHECK_INTERVAL_IF_READY_MILLIS;
-                            for (int j = count; j > 0 && !deploymentHandler.isDeploymentReady(); j--) {   //Parameterize sleep duration hardcoded for 120
-                                try {
-                                    LOGGER.debug("Still deployment is not ready for ExpName {} trail No {}", this.experimentTrial.getExperimentName(), tracker);
-                                    Thread.sleep(EMConstants.TimeConv.DEPLOYMENT_CHECK_INTERVAL_IF_READY_MILLIS);
-                                } catch (InterruptedException e) {
-                                    e.printStackTrace();
-                                }
+                    i -> {
+                        boolean proceedToMetricsCollection = false;
+                        deploymentHandler.initiateDeploy();
+                        //Check if deployment is ready
+                        int deploymentIsReadyWithinMinute = EMConstants.TimeConv.DEPLOYMENT_IS_READY_WITHIN_MINUTE;
+                        long millisMinutes = TimeUnit.MINUTES.toMillis(deploymentIsReadyWithinMinute);
+                        int count = (int)millisMinutes/EMConstants.TimeConv.DEPLOYMENT_CHECK_INTERVAL_IF_READY_MILLIS;
+                        for (int j = count; j > 0 && !deploymentHandler.isDeploymentReady(); j--) {   //Parameterize sleep duration hardcoded for 120
+                            try {
+                                LOGGER.debug("Still deployment is not ready for ExpName {} trail No {}", this.experimentTrial.getExperimentName(), tracker);
+                                Thread.sleep(EMConstants.TimeConv.DEPLOYMENT_CHECK_INTERVAL_IF_READY_MILLIS);
+                            } catch (InterruptedException e) {
+                                e.printStackTrace();
                             }
-                            if (!deploymentHandler.isDeploymentReady())
-                                LOGGER.debug("Giving up for ExpName {} trail No {} for {} attempt", this.experimentTrial.getExperimentName(), this.experimentTrial.getTrialInfo().getTrialNum(), i);
+                        }
```

Comment:
This whole code has to be moved into a function, ideally part of the target abstraction itself

Reply:
I have add an else case just to check the readiness of deployment and proceed to load check. I agree these needs to be part of target abstraction but I feel @msvinaykumar might have a plan/implementation around that part. I would suggest for  a discussion with @msvinaykumar on the plan for this, and if nothing related to that is in current work plan will take it up. I just wanna make sure I'm not duplicating the effort or interrupting an existing plan / work. 

@dinogun , Please lemme know your views on it